src/OpenBinary/reader.jl

```diff
@@ -156,137 +147,157 @@ function PSRI.open(
+        error("'stage_total' must be a positive integer, not '$(stage_total)'")
+    end
 
-    @assert stage_total > 0
-    @assert first_relative_stage == 1 # todo improve this
-    @assert stage_total - first_relative_stage >= 0
+    # TODO: improve this
+    # if first_relative_stage != 1
+    #     @info   first_relative_stage
+    #     @assert false
+    # end
+
+    if !(first_relative_stage <= stage_total)
+        error("'first_relative_stage' must be less than or equal to 'stage_total'")
+    end
 
     if variable_by_series == 0
         # previous versions of the file simply leave the number of scenarios
         # as the number of scenarios in the study
         scenario_total = 1
     end
+
     if !(0 <= variable_by_series <= 1)
-        println(BAD * "variable_by_series = $variable_by_series, expected 0 or 1")
+        @warn("'variable_by_series' must be either 0 or 1, not $(variable_by_series)")
+    end
+
+    if !(scenario_total > 0)
+        error("'scenario_total' must be a positive integer, not $(scenario_total)")
     end
-    @assert scenario_total > 0
-    scenario_exist = variable_by_series == 1
+
+    scenario_exist = (variable_by_series == 1)
 
     if !(0 <= variable_by_block <= 3)
-        println(BAD * "variable_by_block = $variable_by_block, expected 0, 1, 2 or 3")
+        @warn("'variable_by_block' must be 0, 1, 2 or 3, not $(variable_by_block)")
     end
+
     if variable_by_block == 0 && block_total > 1
-        println(BAD * "variable_by_block == 0 but block_total = $block_total")
+        @warn("'variable_by_block' is set to 0 but 'block_total' = $block_total is greater than 1")
     end
-    @assert block_total > 0
-    block_exist = variable_by_block > 0
+    
+    if !(block_total > 0)
+        error("'block_total' must be a positive integer, not $(block_total)")
+    end
+
+    block_exist = (variable_by_block > 0)
 
     if !(0 <= variable_by_hour <= 1)
-        println(BAD * "variable_by_hour = $variable_by_hour, expected 0 or 1")
+        @warn("'variable_by_hour' must be 0, 1, 2 or 3, not $(variable_by_hour)")
     end
-    hours_exist = variable_by_hour == 1
+
+    hours_exist = (variable_by_hour == 1)
 
     if total_agents == 0
-        println("total_agents == 0, nothing to do")
-        # todo
+        @warn("No agents were provided") # todo
     end
+
     @assert name_length > 0
 
     if use_header
         if length(header) > total_agents
             error("Header does not match with expected. Header length = $(length(header)), number ofo agents is $(total_agents)")
         end
-        index = Int[]
-        sizehint!(index, length(header))
-        for agent in header
-            _agent = strip(agent)
-            ret = findfirst(x -> x == _agent, agent_names)
-            if ret === nothing
-                # println("agent $(_agent) not found in file")
-                error("agent $(_agent) not found in file")
+
+        index = sizehint!(Int[], length(header))
```

Comment:
Why is this not `Vector{Int}(undef, lenght(header)`? It should find everyone, if it doesn`t It will error anyway

Reply:
Not exactly... it might be useful to allow for an incomplete header

Reply:
We would still have to fix other things for it to work properly